src/Compilers/CSharp/Test/Emit3/Semantics/ExtensionTests.cs

```diff
@@ -21304,6 +21306,1577 @@ static class E
+            );
+    }
+
+    [Fact]
+    public void NameConflict_29_WithStaticTypeTypeParameter()
+    {
+        var src = @"
+public static class E<T>
+{
+    extension(int p)
+    {
+        public void M1<T>() {}
+    }
+
+    extension<T>(T[] p)
+    {
+        public void M2<T>() {}
+    }
+}
+";
+        var comp = CreateCompilation(src);
+
+        comp.VerifyDiagnostics(
+            // (4,5): error CS9502: Extensions must be declared in a top-level, non-generic, static class
+            //     extension(int p)
+            Diagnostic(ErrorCode.ERR_BadExtensionContainingType, "extension").WithLocation(4, 5),
+            // (6,24): warning CS0693: Type parameter 'T' has the same name as the type parameter from outer type 'E<T>'
+            //         public void M1<T>() {}
+            Diagnostic(ErrorCode.WRN_TypeParameterSameAsOuterTypeParameter, "T").WithArguments("T", "E<T>").WithLocation(6, 24),
+            // (9,5): error CS9502: Extensions must be declared in a top-level, non-generic, static class
+            //     extension<T>(T[] p)
+            Diagnostic(ErrorCode.ERR_BadExtensionContainingType, "extension").WithLocation(9, 5),
+            // (9,15): warning CS0693: Type parameter 'T' has the same name as the type parameter from outer type 'E<T>'
+            //     extension<T>(T[] p)
+            Diagnostic(ErrorCode.WRN_TypeParameterSameAsOuterTypeParameter, "T").WithArguments("T", "E<T>").WithLocation(9, 15),
+            // (11,24): error CS9508: Type parameter 'T' has the same name as an extension container type parameter
+            //         public void M2<T>() {}
+            Diagnostic(ErrorCode.ERR_TypeParameterSameNameAsExtensionTypeParameter, "T").WithArguments("T").WithLocation(11, 24)
+            );
+    }
+
+    [Fact]
+    public void ReceiverParameterScope_01_InStaticMember()
+    {
+        var src = """
+static class Extensions
+{
+    extension(int p)
+    {
+        static int P1 { get => p; }
+        static int M2()
+        {
+            return p;
+        }
+    }
+}
+""";
+        var comp = CreateCompilation(src);
+
+        comp.VerifyDiagnostics(
+            // (5,32): error CS9512: Cannot use extension parameter 'int p' in this context.
+            //         static int P1 { get => p; }
+            Diagnostic(ErrorCode.ERR_InvalidExtensionParameterReference, "p").WithArguments("int p").WithLocation(5, 32),
+            // (8,20): error CS9512: Cannot use extension parameter 'int p' in this context.
+            //             return p;
+            Diagnostic(ErrorCode.ERR_InvalidExtensionParameterReference, "p").WithArguments("int p").WithLocation(8, 20)
+            );
+    }
+
+    [Fact]
+    public void ReceiverParameterScope_02_InStaticMember()
+    {
+        var src = """
+static class Extensions
+{
+    extension(int p)
+    {
+        static int P1
+        {
+            get
+            {
+                int local() => p;
```

Comment:
Do we have a test for the happy case, where the extension parameter is used in a non-static local function or lambda, inside an instance extension member?

Reply:
```Implementation_InstanceMethod_03_WithLocalFunction```, for example?